lib/puppet/util/yaml.rb

```diff
@@ -9,6 +9,48 @@ module Puppet::Util::Yaml
 
   class YamlLoadError < Puppet::Error; end
 
+  def self.disallowed_class(cls)
+    raise YamlLoadError, "Tried to load unspecified class: #{cls}"
+  end
+
+  def self.safe_load(yaml, allowed_classes = [], filename = nil)
+    @psych_safe_load ||= Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.1.0')
+    return YAML.safe_load(yaml, allowed_classes, [], false, filename) if @psych_safe_load
+
+    # emulate needed parts of YAML.safe_load for rubies < 2.1.0
+    data = YAML.parse(yaml, filename)
```

Comment:
Since we ship ruby 2.4.x in puppet-agent, this code path is only used when running puppet as a gem (or bolt) using system ruby. Given that, I'd say let's just drop this and fallback to `YAML.parse`. Otherwise it's technical debt for a ruby version we're trying to move away from.

Reply:
I assume you mean fallback to `YAML.load`.